2-mandatory/3-shopping-list.js

```diff
@@ -20,6 +20,20 @@ let pantry = {
 
 function createShoppingList(recipe) {
   // write code here
+  
+  let missingItems = [];
+
+  for (let i = 0; i < recipe.ingredients.length; i++) {
+    let ingredient = recipe.ingredients[i];
+    if (!pantry.fridgeContents.includes(ingredient) && !pantry.cupboardContents.includes(ingredient)) {
```

Comment:
This looks good! Nice use of ! and && 
--have you thought about concatenating the fridge and cupboard ingredients first?

Reply:
Had problems with this one and checked GPT for help